src/ngraph/runtime/cpu/pass/cpu_layout.cpp

```diff
@@ -268,6 +266,59 @@ void runtime::cpu::pass::CPULayout::set_native_layouts(
     }
 }
 
+static void set_layouts_unaryeltwise(ngraph::runtime::cpu::CPU_ExternalFunction* external_function,
+                                     std::shared_ptr<ngraph::Node> node)
+{
+    auto input_md = mkldnn_utils::get_input_mkldnn_md(node.get(), 0);
+    // Non MKLDNN kernels can handle MKLDNN layouts as long as there are not padded
+    bool md_check = input_md.data.format != mkldnn_format_undef &&
+                    !mkldnn_utils::is_mkldnn_padded_layout(
+                        input_md, ngraph::get_default_order(node->get_input_shape(0)));
+    if (mkldnn_utils::use_mkldnn_kernel(node.get()) || md_check)
+    {
+        vector<memory::desc> o_mds;
+        o_mds.push_back(input_md);
+        set_output_layouts(node, o_mds);
+    }
+    else
+    {
+        set_native_layouts(external_function, node);
+    }
+}
+
+void set_layouts_binaryeltwise(ngraph::runtime::cpu::CPU_ExternalFunction* external_function,
+                               std::shared_ptr<ngraph::Node> node)
+{
+    std::vector<mkldnn::memory::desc> arg_mds{mkldnn_utils::get_input_mkldnn_md(node.get(), 0),
+                                              mkldnn_utils::get_input_mkldnn_md(node.get(), 1)};
+    bool md_check = arg_mds[0].data.format != mkldnn_format_undef &&
+                    arg_mds[1].data.format != mkldnn_format_undef &&
+                    !mkldnn_utils::is_mkldnn_padded_layout(
+                        arg_mds[0], ngraph::get_default_order(node->get_input_shape(0))) &&
+                    !mkldnn_utils::is_mkldnn_padded_layout(
+                        arg_mds[1], ngraph::get_default_order(node->get_input_shape(1)));
+    if (mkldnn_utils::use_mkldnn_kernel(node.get()) || md_check)
+    {
+        vector<memory::desc> i_mds;
+        vector<memory::desc> o_mds;
+        int select = 0;
+        if (std::getenv("NGRAPH_PASS_CPU_LAYOUT_ELTWISE") != nullptr)
+        {
+            const int user_select = std::atoi(std::getenv("NGRAPH_PASS_CPU_LAYOUT_ELTWISE"));
```

Comment:
should we make one of those static so we don't have to recompute it in every compilation?